controllers/openstackdataplanenodeset_controller.go

```diff
@@ -389,3 +411,24 @@ func (r *OpenStackDataPlaneNodeSetReconciler) SetupWithManager(mgr ctrl.Manager)
 			deploymentWatcher).
 		Complete(r)
 }
+
+// GetSpecConfigHash initialises a new struct with only the field we want to check for variances in.
+// We then hash the contents of the new struct using md5 and return the hashed string.
+func (r *OpenStackDataPlaneNodeSetReconciler) GetSpecConfigHash(instance *dataplanev1.OpenStackDataPlaneNodeSet) (string, error) {
+
+	nodeConfig := &NodeConfigElements{
+		NodeTemplate: instance.Spec.NodeTemplate,
+		Nodes:        instance.Spec.Nodes,
+	}
+	jsonBytes, err := json.Marshal(&nodeConfig)
+	if err != nil {
+		return "", err
+	}
+
+	// Hash the values and convert to a slice of bytes required by hex.EncodeToString()
+	var hashSlice []byte
+	hash := md5.Sum(jsonBytes)
+	hashSlice = hash[:]
+
+	return hex.EncodeToString(hashSlice), nil
+}
```

Comment:
Can we use ObjectHash() from lib-common instead?

Reply:
I had no idea this function existed. That is much cleaner, I'll push a new revision.

Reply:
Can we leave this unresolved until it's updated to use lib-common?

Reply:
Yeah, I marked this resolved when I changed it to lib-common last time, but after I realised the configHash wasn't actually changing, I reverted it. In any case, I have changed it again to use lib-common now.

Reply:
thanks. it looks like it's working now based on the passing envtest